pychromecast/socket_client.py

```diff
@@ -252,35 +263,40 @@ def initialize_connection(self):  # pylint:disable=too-many-statements
                     # Resolve the service name. If service is None, we're
                     # connecting directly to a host name or IP-address
                     if service:
+                        if now < retry['next_retry']:
```

Comment:
Why check this here instead of before creating a new socket and outside of the try?

Reply:
Hmm, you're right.. Will fix.